pkg/gds/admin_test.go

```diff
@@ -0,0 +1,939 @@
+	require := s.Require()
+	vasps := []string{
+		"d9da630e-41aa-11ec-9d29-acde48001122",
+		"d9efca14-41aa-11ec-9d29-acde48001122",
+	}
+	WriteVASPs(s, vasps)
+	a := initAdmin(s)
+
+	snippets := []admin.VASPSnippet{
+		{
+			ID:                  vasps[0],
+			Name:                "CharlieBank",
+			CommonName:          "trisa.charliebank.io",
+			RegisteredDirectory: "trisatest.net",
+			VerificationStatus:  pb.VerificationState_SUBMITTED.String(),
+			LastUpdated:         "2021-09-27T04:12:23Z",
+			VerifiedContacts: map[string]bool{
+				"administrative": false,
+				"billing":        true,
+				"legal":          true,
+				"technical":      false,
+			},
+		},
+		{
+			ID:                  vasps[1],
+			Name:                "Delta Assets",
+			CommonName:          "trisa.delta.io",
+			RegisteredDirectory: "trisatest.net",
+			VerificationStatus:  pb.VerificationState_APPEALED.String(),
+			LastUpdated:         "2021-09-18T10:58:22Z",
+			VerifiedContacts: map[string]bool{
+				"administrative": false,
+				"billing":        true,
+				"legal":          true,
+				"technical":      false,
+			},
+		},
+	}
+
+	// List all VASPs on the same page
+	request := &apiRequest{
+		fn:     a.ListVASPs,
+		method: http.MethodGet,
+		path:   "/v2/vasps",
+	}
+	actual := &admin.ListVASPsReply{}
+	rep := s.doRequest(request, actual)
+	require.Equal(http.StatusOK, rep.StatusCode)
+	require.Equal(len(vasps), actual.Count)
+	require.Equal(1, actual.Page)
+	require.Equal(100, actual.PageSize)
+	require.Len(actual.VASPs, len(snippets))
+	sort.Slice(actual.VASPs, func(i, j int) bool {
+		return actual.VASPs[i].ID < actual.VASPs[j].ID
+	})
+	require.Equal(snippets, actual.VASPs)
+
+	// List VASPs with an invalid status
+	request.path = "/v2/vasps?status=invalid"
+	rep = s.doRequest(request, nil)
+	require.Equal(http.StatusBadRequest, rep.StatusCode)
+
+	// List VASPs with the specified status
+	request.path = "/v2/vasps?status=" + snippets[0].VerificationStatus
+	rep = s.doRequest(request, actual)
+	require.Equal(http.StatusOK, rep.StatusCode)
+	require.Equal(len(vasps), actual.Count)
+	require.Equal(1, actual.Page)
+	require.Equal(100, actual.PageSize)
+	require.Len(actual.VASPs, 1)
+	require.Equal(snippets[0], actual.VASPs[0])
+
+	// List VASPs on multiple pages
+	request.path = "/v2/vasps?page=1&page_size=1"
+	rep = s.doRequest(request, actual)
+	require.Equal(http.StatusOK, rep.StatusCode)
+	require.Equal(len(vasps), actual.Count)
+	require.Equal(1, actual.Page)
+	require.Equal(1, actual.PageSize)
+	require.Len(actual.VASPs, 1)
+	require.Equal(snippets[0], actual.VASPs[0])
+
+	request.path = "/v2/vasps?page=2&page_size=1"
+	rep = s.doRequest(request, actual)
+	require.Equal(http.StatusOK, rep.StatusCode)
+	require.Equal(len(vasps), actual.Count)
+	require.Equal(2, actual.Page)
+	require.Equal(1, actual.PageSize)
+	require.Len(actual.VASPs, 1)
+	require.Equal(snippets[1], actual.VASPs[0])
+
+	request.path = "/v2/vasps?page=3&page_size=1"
```

Comment:
Excellent series of tests. For future proofing we may want to consider a testTable with test case loop, just so it's easy to adapt. Perhaps we can create a story for refactoring?